web/packages/teleterm/src/ui/DocumentGateway/useDocumentGateway.ts

```diff
@@ -127,12 +114,64 @@ export function useDocumentGateway(doc: types.DocumentGateway) {
     connected,
     reconnect: createGateway,
     connectAttempt,
+    disconnectAttempt,
+    changeTargetSubresourceName,
+    changeTargetSubresourceNameAttempt,
+    changePort,
+    changePortAttempt,
+  };
+}
+
+//TODO(gzdunek): Refactor DocumentGateway so the hook below is no longer needed.
+// We should move away from using one big hook per component.
+export function useDocumentGateway(doc: types.DocumentGateway) {
+  const { documentsService: workspaceDocumentsService } = useWorkspaceContext();
+
+  const {
+    gateway,
+    reconnect,
+    connectAttempt,
+    disconnectAttempt,
+    disconnect,
+    connected,
+    changePort,
+    changePortAttempt,
+    changeTargetSubresourceNameAttempt,
+    changeTargetSubresourceName,
+    defaultPort,
+  } = useGateway(doc);
+
+  const runCliCommand = () => {
+    if (!isDatabaseUri(doc.targetUri)) {
+      return;
+    }
```

Comment:
`DocumentGatewayCliClient` expects `targetUri` to be only `DatabaseUri`, while in `DocumentGateway` it can be both `DatabaseUri` and `AppUri`. If I don't narrow down the type, I get a TS error.